lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -161,6 +168,44 @@ defmodule Bamboo.SMTPAdapter do
     |> add_smtp_line(text_body)
   end
 
+  defp add_multipart_mixed_split(body, delimiter) do
+    body
+    |> add_smtp_line(delimiter)
+  end
+
+  defp add_attachment_header(body, attachment) do
+    << random :: size(32) >> = :crypto.strong_rand_bytes(4)
+    body
+    |> add_smtp_line("Content-Type: attachment.content_type; name=\"#{attachment.filename}")
```

Comment:
I replied to your question but as the discussion has been collapsed by GitHub, you probably missed it. So, I add it back here. :)

> Ghouli:
> Sorry, my english fails here, do you mean that we should check if attachment.content_type is set before using? If so, we won't have to do that here. Bamboo.Attachment already sets content_type to "application/octet-stream" if not set on creation. If you meant something else, please let me now and I'll fix it. Thanks for the suggestions! :)

Sorry if my comment wasn't clear enough :)

I meant we should use the value of the attachment.content_type variable instead of the literal string "attachement.content_type":

```diff
- |> add_smtp_line("Content-Type: attachment.content_type; name=\"#{attachment.filename}\"")
+ |> add_smtp_line("Content-Type: #{attachment.content_type}; name=\"#{attachment..filename}\"")
```